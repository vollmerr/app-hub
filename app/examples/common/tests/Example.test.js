

import React from 'react';
import { shallow } from 'enzyme';

import Example from '../Example';

const header = 'test header';
const desc = ['test desc 1', 'test desc 2'];
const Children = () => <div>test children</div>;

const wrapper = shallow(
  <Example header={header} desc={desc}>
    <Children />
  </Example>
);


describe('<Example />', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render its children', () => {
    expect(wrapper.find(Children).length).toEqual(1);
  });

  it('should render a header', () => {
    expect(wrapper.find('h1').length).toEqual(1);
  });

  it('should render a description', () => {
    expect(wrapper.find('p').length).toEqual(desc.length);
  });
});
