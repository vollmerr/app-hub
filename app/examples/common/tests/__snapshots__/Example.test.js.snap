// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Example /> should render correctly 1`] = `
<Example__Wrapper>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      test header
    </title>
  </HelmetWrapper>
  <Example__Left>
    <h1>
      test header
    </h1>
    <p
      key="0"
    >
      test desc 1
    </p>
    <p
      key="1"
    >
      test desc 2
    </p>
  </Example__Left>
  <Example__Right>
    <Children />
  </Example__Right>
</Example__Wrapper>
`;
